### DISCUSSION TITLE
Remix ErrorBoundary on Vercel

### DISCUSSION THREAD

**Discussion by sergical** [remix-run/remix#7757] [Q&A] [answered]
### Summary

The `ErrorBoundary` route gets built when you have any Edge functions in your app as an Edge function.

`main`: https://github.com/sergical/remix-playground/blob/main/app/routes/edge.tsx

<img width="1219" alt="Screenshot 2023-10-24 at 5 05 58 PM" src="https://github.com/vercel/community/assets/3760543/4388dbe1-9e2e-4786-a2f2-ac798516f553">

`not-edge`: https://github.com/sergical/remix-playground/blob/not-edge/app/routes/edge.tsx

<img width="1209" alt="Screenshot 2023-10-24 at 5 12 39 PM" src="https://github.com/vercel/community/assets/3760543/0322f86d-00c6-4df8-8d63-53d146bed271">

The actual `Edge` Error boundary works:

https://remix-playground.vercel.app/edge?error=error

<img width="1680" alt="Screenshot 2023-10-24 at 5 14 30 PM" src="https://github.com/vercel/community/assets/3760543/adf4bdfe-15dc-4e6e-8863-f265bf3cdb77">

However the root level error boundary crashes: 

<img width="1680" alt="Screenshot 2023-10-24 at 5 14 21 PM" src="https://github.com/vercel/community/assets/3760543/707d674c-bf70-4cb9-b14c-d405a645c17a">

And it doesn't crash on the `not-edge` branch:

https://remix-playground-git-not-edge-sergiy.vercel.app/adsf

<img width="1680" alt="Screenshot 2023-10-24 at 5 26 39 PM" src="https://github.com/vercel/community/assets/3760543/343a01fe-9dd9-4363-b231-182e12098d01">

I understand that root `ErrorBoundary` is what gets fed into the `404` Vercel function. But I was hoping we could have some flexibility on it not being `Edge` when it doesn't need to be. For example, each `edge` file has it's own ErrorBoundary so the root 404 can be built as Serverless. Otherwise all the default error handling that remix provides does not work and it becomes a bad DX and UX :sad

### Example

https://remix-playground.vercel.app/

### Steps to Reproduce

1. Clone https://github.com/sergical/remix-playground
2. Deploy to Vercel
3. Notice 404 being built as Edge
4. Create separate deployment similar to https://github.com/sergical/remix-playground/blob/main/app/routes/edge.tsx#L4
5. Notice 404 being built as Serverless

**Comment by sergiodxa** [comment, marked as answer]
This seems like a problem with Vercel, this split of routes so some are Edge and others are Serverless is something Vercel does when you deploy to them, not something Remix supports out of the box.

**Reply by sergical** [reply]
Yup, I started a thread there: https://github.com/orgs/vercel/discussions/4596 This was just a sanity check and for visibility, thank you! ❤️
